File: app/src/main/java/com/egecius/rxjava2_demo_2/rx/EgisException.java
package com.egecius.rxjava2_demo_2.rx;

public class EgisException extends Exception {

    private final String message;

    public EgisException() {
        message = null;
    }

    public EgisException(String message) {
        super(message);
        this.message = message;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof EgisException)) return false;

        EgisException that = (EgisException) o;

        return message != null ? message.equals(that.message) : that.message == null;
    }

    @Override
    public int hashCode() {
        return message != null ? message.hashCode() : 0;
    }
}

File: app/src/main/java/com/egecius/rxjava2_demo_2/rx/EgisException.kt
package com.egecius.rxjava2_demo_2.rx

class EgisException(message: String? = null) : Exception(message) {

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is EgisException) return false
        return true
    }

}